ksqldb-engine/src/main/java/io/confluent/ksql/physical/scalablepush/ScalablePushRegistry.java

```diff
@@ -364,4 +457,129 @@ private void runLatest(final LatestConsumer latestConsumerToRun) {
   private String getLatestConsumerGroupId() {
     return sourceApplicationId + LATEST_CONSUMER_GROUP_SUFFIX;
   }
+
+  private synchronized void catchupAssignmentUpdater(
+      final Collection<TopicPartition> newAssignment
+  ) {
+    for (CatchupConsumer catchupConsumer : catchupConsumers.values()) {
+      catchupConsumer.newAssignment(newAssignment);
+    }
+  }
+
+  /**
+   * Creates the latest consumer and its underlying kafka consumer.
+   * @param processingQueue The queue on which to send an error if anything goes wrong
+   * @return The new LatestConsumer
+   */
+  private CatchupConsumer createCatchupConsumer(
+      final ProcessingQueue processingQueue,
+      final PushOffsetRange offsetRange,
+      final String consumerGroup
+  ) {
+    KafkaConsumer<Object, GenericRow> consumer = null;
+    CatchupConsumer catchupConsumer = null;
+    try {
+      consumer = kafkaConsumerFactory.create(
+          ksqlTopic, logicalSchema, serviceContext, consumerProperties, ksqlConfig,
+          consumerGroup);
+      catchupConsumer = catchupConsumerFactory.create(ksqlTopic.getKafkaTopicName(), isWindowed(),
+          logicalSchema, consumer, latestConsumer::get, catchupCoordinator, offsetRange,
+          Clock.systemUTC());
+      return catchupConsumer;
+    } catch (Exception e) {
+      LOG.error("Couldn't create catchup consumer", e);
+      processingQueue.onError();
+      // We're not supposed to block here, but if it fails here, hopefully it can immediately close.
+      if (consumer != null) {
+        consumer.close();
+      }
+      throw e;
+    }
+  }
+
+  /**
+   * Starts the catchup consumer
+   * @param processingQueue The queue to register with the catchup consumer.
+   * @param catchupMetadata The catchup metadata
+   */
+  private synchronized void startCatchup(
+      final ProcessingQueue processingQueue,
+      final CatchupMetadata catchupMetadata
+  ) {
+    final CatchupConsumer catchupConsumer = createCatchupConsumer(processingQueue,
+        catchupMetadata.getPushOffsetRange(),
+        catchupMetadata.getCatchupConsumerGroup());
+    catchupConsumer.register(processingQueue);
```

Comment:
super nit: seems like this should just be in `createcatchupConsumer`, no?

Reply:
I'm just doing normal camelCase.  Would you normally not capitalize in this case?